core/src/main/scala/org/apache/spark/util/AccumulatorV2.scala

```diff
@@ -136,15 +181,92 @@ abstract class AccumulatorV2[IN, OUT] extends Serializable {
   def reset(): Unit
 
   /**
+   * Takes the inputs and accumulates. e.g. it can be a simple `+=` for counter accumulator.
+   * Developers should extend addImpl to customize the adding functionality.
+   */
+  final def add(v: IN): Unit = {
```

Comment:
oh crap.  you are going to hate me ... making this `final` is going to break binary compatibility.

is there any other alternative? I will keep thinking but I can't come up with anything.